tests/test_cases/test_cocotb/test_cocotb.py

```diff
@@ -712,6 +712,58 @@ def test_binary_value_compat(dut):
     yield Timer(100)  # Make it do something with time
 
 
+@cocotb.test()
+def join_finished(dut):
+    """
+    Test that joining a coroutine that has already been joined gives
+    the same result as it did the first time.
+    """
+
+    retval = None
+
+    @cocotb.coroutine
+    def some_coro():
+        yield Timer(1)
+        raise ReturnValue(retval)
+
+    coro = cocotb.fork(some_coro())
+
+    retval = 1
+    x = yield coro.join()
+    assert x == 1
```

Comment:
Almost all other tests use "raise TestFailure" instead of asserts in tests. I personally find asserts much nicer, but maybe we should stay consistent here? (That's not a must, I'd love if we can switch to asserts fully)